cobalt/user_agent_string.h

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COBALT_USER_AGENT_STRING_H_
+#define COBALT_USER_AGENT_STRING_H_
+
+#include "cobalt/user_agent_platform_info.h"
+
+namespace cobalt {
+
+// This function is deterministic and non-dependent on global variables and
+// state.  It is dependent only on |platform_info|.
+std::string CreateUserAgentString(const UserAgentPlatformInfo& platform_info);
```

Comment:
I think it's overkill to have a full file just for this free function. Could we fold it and make it `UserAgentPlatformInfo::ToString()` ?

Reply:
Yea agreed, move the logic to `UserAgentPlatformInfo::ToString()`.